### PR TITLE
En 957 role endpoints

### PR DESCRIPTION
**Pull request by stefanopuddu**
add roles endpoint 
add  /permission [GET]

### CODE REVIEW

File: engine/src/main/java/org/entando/entando/web/role/RoleController.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    @RequestMapping(value = "/{roleCode}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<RestResponse> getRole(@PathVariable String roleCode) {
+        RoleDto role = this.getRoleService().getRole(roleCode);
+        return new ResponseEntity<>(new RestResponse(role), HttpStatus.OK);
+    }
+
+    @RestAccessControl(permission = Permission.SUPERUSER)
+    @RequestMapping(value = "/{roleCode}/references", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<RestResponse> getRoleReferences(@PathVariable String roleCode, RestListRequest requestList) {
+        PagedMetadata<UserDto> result = this.getRoleService().getRoleReferences(roleCode, requestList);
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/role/RoleController.java]: Add logging.

File: engine/src/main/java/org/entando/entando/web/role/RoleController.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        return new ResponseEntity<>(new RestResponse(result.getBody(), null, result), HttpStatus.OK);
+    }
+
+    @RestAccessControl(permission = Permission.SUPERUSER)
+    @RequestMapping(value = "/{roleCode}", method = RequestMethod.PUT, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<RestResponse> updateRole(@PathVariable String roleCode, @Valid @RequestBody RoleRequest roleRequest, BindingResult bindingResult) {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/role/RoleController.java]: Add logging

File: engine/src/main/java/org/entando/entando/web/role/RoleController.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+            throw new ValidationGenericException(bindingResult);
+        }
+
+        RoleDto role = this.getRoleService().updateRole(roleRequest);
+        return new ResponseEntity<>(new RestResponse(role), HttpStatus.OK);
+    }
+
+    @RestAccessControl(permission = Permission.SUPERUSER)
+    @RequestMapping(method = RequestMethod.POST, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<RestResponse> addRole(@Valid @RequestBody RoleRequest roleRequest, BindingResult bindingResult) throws ApsSystemException {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/role/RoleController.java]: Add logging.

File: engine/src/main/java/com/agiletec/aps/system/services/role/IRoleManager.java [hunk 4]
```diff
@@ -24,93 +24,95 @@
...
+
+    /**
+     * Aggiunge un permesso di autorizzazione.
+     *
+     * @param permission Il permesso da aggiungere.
+     * @throws ApsSystemException in caso di errore.
+     */
+    public void addPermission(Permission permission) throws ApsSystemException;
+
+    //public int getRoleUses(Role role) throws ApsSystemException;
```
Comment (joewhite101) [inline engine/src/main/java/com/agiletec/aps/system/services/role/IRoleManager.java]: remove.

File: engine/src/main/java/org/entando/entando/aps/system/services/role/RoleDdoBuilder.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+import java.util.List;
+
+import com.agiletec.aps.system.services.role.IRoleManager;
+import com.agiletec.aps.system.services.role.Role;
+import org.entando.entando.aps.system.services.DtoBuilder;
+import org.entando.entando.aps.system.services.role.model.RoleDto;
+
+
+public class RoleDdoBuilder extends DtoBuilder<Role, RoleDto> {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/role/RoleDdoBuilder.java]: Typo in name "Ddo" --> "Dto"

File: engine/src/main/java/org/entando/entando/web/permission/validator/PermissionValidator.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+
+    @Override
+    public boolean supports(Class<?> paramClass) {
+        return PermissionRequest.class.equals(paramClass);
+    }
+
+    @Override
+    public void validate(Object target, Errors errors) {
+        //
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/permission/validator/PermissionValidator.java]: remove

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Pull request merged**